### PR TITLE
unify macros and cmake options that control the suppression of deprecation warnings

### PR DESCRIPTION
**Pull request by ericniebler**
## Description

as part of the theme of unifying libcu++, cub, and thrust into a single library, all of the public `CUB_`, `THRUST_`, and `LIBCUDACXX_` config macros and cmake options must be migrated to `CCCL_`. this PR does that for the macros and cmake options that are used to suppress deprecation warnings.

the old macros and cmake options are still supported. a use of a `CUB_` or `THRUST_` cmake option will cause a warning at config time about the name change.

the definition of any `CUB_`, `THRUST_`, or `LIBCUDACXX_` macro or config option will enable the corresponding `CCCL_` option, which in turn controls deprecation suppressions for all of CCCL.

## Checklist

- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by pciolkosz — APPROVED** [review]

File: cub/cub/util_deprecated.cuh [hunk 1]
```diff
@@ -1,96 +0,0 @@
...
-#include <cub/detail/type_traits.cuh>
-#include <cub/util_debug.cuh>
-
-#if defined(THRUST_IGNORE_DEPRECATED_API) && !defined(CUB_IGNORE_DEPRECATED_API)
-#  define CUB_IGNORE_DEPRECATED_API
-#endif
-
-#ifdef CUB_IGNORE_DEPRECATED_API
-//! deprecated [Since 2.8]
-#  define CUB_DEPRECATED
```
Comment (miscco) [inline cub/cub/util_deprecated.cuh:54]: Those were user facing macros that we need to preserve

File: thrust/thrust/detail/config/deprecated.h [hunk 2]
```diff
@@ -1,70 +0,0 @@
...
-#include <thrust/detail/config/compiler.h>
-#include <thrust/detail/config/cpp_dialect.h>
-
-#if defined(CUB_IGNORE_DEPRECATED_API) && !defined(THRUST_IGNORE_DEPRECATED_API)
-#  define THRUST_IGNORE_DEPRECATED_API
-#endif
-
-#ifdef THRUST_IGNORE_DEPRECATED_API
-//! deprecated [Since 2.8]
-#  define THRUST_DEPRECATED
```
Comment (miscco) [inline thrust/thrust/detail/config/deprecated.h:42]: Ditto those were user facing

File: cub/cub/util_deprecated.cuh (hunk 1 above)
Comment (ericniebler) [reply cub/cub/util_deprecated.cuh:54]: i found no uses of `CUB_DEPRECATED` on github that were not in clones of the CUB or CCCL repos.

File: thrust/thrust/detail/config/deprecated.h (hunk 2 above)
Comment (ericniebler) [reply thrust/thrust/detail/config/deprecated.h:42]: i found exactly one use of `THRUST_DEPRECATED` that is not in a clone of the Thrust or CCCL repos. i can add it back if you think it's worth it.

**Pull request merged**